src/vector/layer.rs

```diff
@@ -330,6 +335,32 @@ impl<'a> Layer<'a> {
             gdal_sys::OGR_L_ResetReading(self.c_layer);
         }
     }
+
+    /// Set a new attribute query that restricts features when using the feature iterator.
+    ///
+    /// Parameters:
+    /// - `psz_query` in restricted SQL WHERE format
+    ///
+    pub fn set_attribute_filter(&self, psz_query: &str) -> Result<()> {
```

Comment:
I _think_ the other methods use Rust parameter naming conventions, so maybe we can call this `query` instead.

Reply:
05c99b5